dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/authentication/AbstractFlow.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.security.oidcclientcore.authentication;
+
+import io.openliberty.security.oidcclientcore.client.OidcClientConfig;
+
+public abstract class AbstractFlow implements Flow {
+
+    public static final String AUTHORIZATION_CODE_RESPONSE_TYPE = "code";
+    public static final String IMPLICIT_RESPONSE_TYPE = "token";
+
+    public static Flow getInstance(OidcClientConfig oidcClientConfig) {
+        String configResponseType = oidcClientConfig.getResponseType();
+        String[] responseTypes = configResponseType.split(" ");
+        for (String responseType : responseTypes) {
+            if (IMPLICIT_RESPONSE_TYPE.equals(responseType)) {
```

Comment:
Should implicitly check for "id_token" or "id_token token". Otherwise, current impl will match Hybrid's types such as "code token" and "code id_token token", which will not be supported.

Reply:
Will take care of in my next delivery 👍